### PR TITLE
[IMP] Some improves to make more fast the methods

### PR DESCRIPTION
**Pull request by luistorresm**

### CODE REVIEW

**Comment by luistorresm** [comment]
@hbto 

What do you think about this PR?

Regards

File: aging_due_report/wizard/wizard.py [hunk 1]
```diff
@@ -113,6 +113,7 @@ def _get_amount_span(self, field_names, res_line, doc, spans, direction):
                  'document_ids.payment', 'document_ids.total')
     def _get_amount(self):
         field_sum = ['residual', 'payment', 'total', 'not_due']
+        field_sum2 = ['residual', 'payment', 'total']
```
Comment (hbto) [inline aging_due_report/wizard/wizard.py]: field_sum = ['residual', 'payment', 'total']
...
field_names = field_spans + field_sum + ['not_due']

File: aging_due_report/wizard/wizard.py [hunk 2]
```diff
@@ -123,9 +124,8 @@ def _get_amount(self):
             for fn in field_names:
                 res[fn] = 0.0
             for doc in record.document_ids:
-                for fsum in field_sum:
-                    if fsum != 'not_due':
-                        res[fsum] += doc.read([fsum])[0][fsum]
+                for fsum in field_sum2:
```
Comment (hbto) [inline aging_due_report/wizard/wizard.py]: keeping same field_sum:
getting rid of conditional statement

**Comment by hbto** [comment]
@luistorresm Thanks for your optimization,
Just a minor changes in form, and all can be merged.

Regards

**Comment by luistorresm** [comment]
Done @hbto 

Thanks

**Comment by hbto** [comment]
+1 Can be merged

**Comment by luistorresm** [comment]
@dsabrinarg 

Could you merge this PR please

File: aging_due_report/wizard/wizard.py [hunk 3]
```diff
@@ -302,14 +301,14 @@ def _get_lines_by_partner_without_invoice(self, rp_ids):
 
         moves_invoice_ids = [aawd_brw.invoice_id.move_id.id
                              for aawp_brw in wzd_brw.partner_ids
-                             for aawd_brw in aawp_brw.document_ids
-                             if aawd_brw.invoice_id]
+                             for aawd_brw in aawp_brw.document_ids.filtered(
+                                 lambda r: r.invoice_id)]
```
Comment (moylop260) [inline aging_due_report/wizard/wizard.py]: You can use `filtered('invoice_id')`

**Review by moylop260 — CHANGES_REQUESTED** [review]

File: aging_due_report/wizard/wizard.py [hunk 4]
```diff
@@ -302,14 +301,14 @@ def _get_lines_by_partner_without_invoice(self, rp_ids):
...
 
         items_invoice_ids = [
             aml_brw.id for aawp_brw in wzd_brw.partner_ids
-            for aawd_brw in aawp_brw.document_ids
-            for aml_brw in aawd_brw.invoice_id.payment_ids
-            if aawd_brw.invoice_id]
+            for aawd_brw in aawp_brw.document_ids.filtered(
+                lambda r: r.invoice_id)
+            for aml_brw in aawd_brw.invoice_id.payment_ids]
```
Comment (moylop260) [inline aging_due_report/wizard/wizard.py:309]: Is posible use mapped?

File: aging_due_report/wizard/wizard.py [hunk 5]
```diff
@@ -347,26 +346,27 @@ def _get_lines_by_partner_without_invoice(self, rp_ids):
...
                 date_due = (date_due and min(date_due) or
                             min([amy_brw.date_maturity or amy_brw.date
                                  for amy_brw in aml_lines]))
                 date_emission = [amz_brw.date
-                                 for amz_brw in aml_lines
-                                 if amz_brw.journal_id.type in (
-                                     'sale', 'purchase')]
+                                 for amz_brw in aml_lines.filtered(
+                                     lambda r: r.journal_id.type in (
+                                         'sale', 'purchase'))]
```
Comment (moylop260) [inline aging_due_report/wizard/wizard.py:356]: Is posible use mapped here too?

Comment (luistorresm) [reply aging_due_report/wizard/wizard.py:356]: @moylop260 

How do you think that the mapped method is used here?

Comment (moylop260) [reply aging_due_report/wizard/wizard.py:356]: You are using filtered (good!) but mapped is not used yet.

I mean, you could use:
`aml_lines.filtered(lambda r: r.journal_id.type in ('sale', 'purchase'))].mapped('date')`

**Comment by luisg123v** [comment]
This seems abandoned, it has no activity from a long time ago, so I'll close it.

Feel free to reopen if if it's not.

**Pull request closed without merging**